### DISCUSSION TITLE
Hay and Davies Modification for infinite_sheds

### DISCUSSION THREAD

**Discussion by john-moseley-arraytech** [pvlib/pvlib-python#1551] [Q&A] [answered]
Hello,

I am interested in modifying the infinite sheds bifacial model to incorporate the Hay and Davies sky diffuse model. I believe all I need to modify is the _poa_sky_diffuse_pv function for Hay and Davies arguments and expressions, e.g., as shown below. (There's also a trivial modification to the get_irradiance_poa function to account for the new arguments in _poa_sky_diffuse_pv.) The results I've obtained with this modification make sense, but I've only considered simple test cases. Do other functions in infinite_sheds require modifications to account for Hay and Davies as well?

If the Hay and Davies modifications to infinite_sheds would be of interest to the community, I'd be happy to try to incorporate them, considering compatibility with previous versions and other conventions.

Regards,

John Moseley

![image](https://user-images.githubusercontent.com/102536311/190030236-0749558d-7abf-4e24-897a-d24dbda1a686.png)

**Comment by mikofski** [comment]
Hi John, I think this would be a super improvement. Would it be possible to have Hay-Davies be an option? IE: add a `poa_method` argument and set it to `None` by default, but allow `hay-davies` as an option to apply the methods you are suggesting? Also, is any of the code you suggest already exist in the Hay-Davies transposition functions? Would be nice if we can reorganize code to reuse that, then we wouldn't have to have multiple citations and also easier for testing and code maintenance maybe? Just some ideas, sorry if I'm making it too complicated. Thanks!

**Reply by john-moseley-arraytech** [reply]
Thanks for your reply, Mark. I was thinking of adding something along the lines of a poa_method argument as you suggest--it seems like the most flexible approach. I may ask for your help there, assuming this moves forward as an issue. The code I used in the modification to _poa_sky_diffuse_pv function is directly from the pvlib.irradiance.haydavies function. However, I did not apply the vf integration terms (vf_shade_sky_integ and vf_noshade_sky_integ) to the circumsolar term in haydavies. This is related to Cliff's comment below, and I'll reply there as well.

**Comment by cwhanse** [comment, marked as answer]
My concern is that the calculations in `infinite_sheds` that use view factors, e.g., reflections from the ground to the rear surface, implicitly assume that the incoming irradiance is isotropic. There's no weighting or other accommodation in the view factor to account for non-isotropic sources.

The Hay-Davies model differs from the simple, isotropic sky model, only by distinguishing circumsolar irradiance from the rest of sky diffuse irradiance. The output of the pvlib `hay-davies` function is the sum of both quantities. It may be reasonable to combine Hay-Davies sky diffuse with the isotropic view factors, but it's important to acknowledge the inconsistency.

One could create a work-around by creating a variant of Hay-Davies that returns the circumsolar and rest-of-sky irradiance separately, adding the circumsolar to direct irradiance, and using rest-of-sky as DHI.

**Reply by john-moseley-arraytech** [reply]
Thanks, Cliff. This is helpful.

So the workaround I was thinking of is shown in the screenshot above, where I apply the vf integration terms (vf_shade_sky_integ and vf_noshade_sky_integ) to the rest-of-sky irradiance term from haydavies but not the circumsolar term. Is that technically correct, or no? Would I need similar workarounds for other functions besides _poa_sky_diffuse_pv in infinite_sheds?

**Reply by cwhanse** [reply]
I don't think that's correct. The first term (DHI x AI x Rb) is the circumsolar irradiance received directly on the tilted plane, shaded and unshaded; I think it should be multiplied by f_x so that circumsolar falls only on the unshaded portion of the tilted plane. The other term (DHI x (1 - AI) x (weighted sum of view factors) is the rest-of-sky diffuse that is received on both shaded and unshaded portions.

This may be half of the needed modification. The other portion would be in the calculation of ground-reflected irradiance received on the tilted plane, which assumes that the incoming diffuse irradiance (circumsolar and rest-of-sky) to the ground is isotropic. That's implemented in a sequence of private functions starting [here](https://github.com/pvlib/pvlib-python/blob/ce7ddbcd328f8aad391e2db5aa32e6acc5dfd272/pvlib/bifacial/infinite_sheds.py#L541). 

I think it would easier (and more explainable) to use `get_irradiance_poa` with dhi = rest-of-sky diffuse, and dni = DNI + circumsolar.

**Reply by cwhanse** [reply]
It's a good question to ask: why is circumsolar not already in direct? My opinion, because measured DNI excludes circumsolar, because (in turn) the instruments that measure DNI are designed that way.

**Reply by john-moseley-arraytech** [reply]
Much appreciated, Cliff. If I understand correctly, you're suggesting to just use get_irradiance_poa as is but with the dhi = rest-of-sky diffuse and dni = DNI + circumsolar, where rest-of-sky diffuse and circumsolar components are determined from the Hay and Davies sky diffuse model.

Perhaps that's still worth building into the functionality of infinite_sheds? E.g., have an argument for poa_method as Mark suggested, and then if the method is 'haydavies', apply/transform dhi = rest-of-sky diffuse and dni = DNI + circumsolar?

**Reply by kandersolar** [reply]
For reference, this functionality was implemented in #1668 and will be included in the next release (0.9.5).
